File: be_fsharp_dotnet_core/TodoList/Controllers/TodoController.fs
namespace TodoList.Controllers

open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Configuration
open System.Collections.Generic
open System.Linq
open System.Threading.Tasks
open Dapper
open Npgsql
open Todolist.Models

[<ApiController>]
[<Route("/tasks")>]
type TodoController (configuration:IConfiguration) =
    inherit ControllerBase()
    let _connectionString = configuration.GetConnectionString "DefaultConnection"

    [<HttpGet>]
    member __.Get() : Task<TaskList> =
        async {            
            let connection =  new NpgsqlConnection(_connectionString)
            do! Async.AwaitTask (connection.OpenAsync())
            let! results = Async.AwaitTask (connection.QueryAsync<TodoTask>("Select * FROM tasks ORDER BY priority asc"))
            let tasks = {
                Tasks=List.ofSeq results;
                Position=0;
                Length=results.Count()
            }
            do! Async.AwaitTask (connection.CloseAsync())
            return tasks
        } |> Async.StartAsTask

    [<HttpPost>]
    member __.Post(task: TodoTask) : Task<Dictionary<string,obj>> =
        async {
            let connection =  new NpgsqlConnection(_connectionString)
            do! Async.AwaitTask (connection.OpenAsync())
            let sql = @"INSERT into tasks (text, priority) VALUES (@Text, @Priority)";
            Async.AwaitTask (connection.ExecuteAsync(sql, task)) |> Async.RunSynchronously |> ignore
            let task = Dictionary<string, obj>()
            task.Add("task", task)
            do! Async.AwaitTask (connection.CloseAsync())
            return task
        } |> Async.StartAsTask


    [<HttpPut>]
    member __.Put(task: TodoTask) : Task<Dictionary<string,obj>> =
        async {
            let connection =  new NpgsqlConnection(_connectionString)
            do! Async.AwaitTask (connection.OpenAsync())
            let sql = @"UPDATE tasks SET (text = @Text, priority = @Priority) WHERE id = @Id";
            Async.AwaitTask (connection.ExecuteAsync(sql, task)) |> Async.RunSynchronously |> ignore
            let task = Dictionary<string, obj>()
            task.Add("task", task)
            do! Async.AwaitTask (connection.CloseAsync())
            return task
        } |> Async.StartAsTask

    [<HttpDelete>]
    member __.Delete() =
        let connection =  new NpgsqlConnection(_connectionString)
        connection.Execute("DELETE FROM tasks;")

[<ApiController>]
[<Route("/status")>]
type StatusController () =
    inherit ControllerBase()

    [<HttpGet>]
    member __.Get() : Task<string> =
        async {
            return "ok"
        } |> Async.StartAsTask

File: be_csharp_dotnet/Controllers/TodoController.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using TodoList.Models;

namespace TodoList.Controllers
{
    [ApiController]
    [Route( "[controller]" )]
    public class TodoController : ControllerBase
    {
        private readonly IRepo _repo;

        public TodoController( IRepo repo ) =>
            _repo = repo;

        [HttpDelete]
        public async Task<IActionResult> Delete()
        {
            await _repo.Delete();
            return NoContent();
        }

        [HttpGet]
        public async Task<IActionResult> Get( Int32? page )
        {
            var tasks = await _repo.Get( page );
            return Ok( tasks );
        }

        [HttpPost]
        public async Task<IActionResult> Post( TodoTask task )
        {
            await _repo.Add( task );

            return Ok( new Dictionary<String, TodoTask>
            {
                { "task", task }
            } );
        }

        [HttpPut]
        public async Task<IActionResult> Put( TodoTask task )
        {
            await _repo.Update( task );

            return Ok( new Dictionary<String, TodoTask>
            {
                { "task", task }
            } );
        }
    }
}